crates/forge-runner/src/test_execution_syscall_handler.rs

```diff
@@ -82,6 +83,30 @@ impl<'a> TestExecutionSyscallHandler<'a> {
             run_resources: RunResources::default(),
         }
     }
+
+    pub fn context(&self) -> &EntryPointExecutionContext {
+        self.child.child.child.context
+    }
+
+    pub fn get_all_execution_resources(&mut self) -> ExecutionResources {
+        let mut all_resources = ExecutionResources::default();
+        all_resources.vm_resources += &self.child.child.child.resources.vm_resources;
+        all_resources.vm_resources += &self.child.child.cheatnet_state.used_resources.vm_resources;
+
+        all_resources
+            .syscall_counter
+            .extend(&self.child.child.child.resources.syscall_counter);
+        all_resources.syscall_counter.extend(
+            &self
+                .child
+                .child
+                .cheatnet_state
+                .used_resources
+                .syscall_counter,
+        );
+
+        all_resources
+    }
```

Comment:
Those shouldn't be  methods of a syscall handler, it is not a purpose of this type. Imo we should make them standalone functions (this one is important!)